docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+    "job_id": "7446098166",
+    "ref": "main",
+    "ref_type": "branch",
+    "ref_path": "refs/heads/main",
+    "ref_protected": "true",
+    "groups_direct": [
+        "openpubkey"
+    ],
+    "runner_id": 12270852,
+    "runner_environment": "gitlab-hosted",
+    "sha": "9898863c7dcd844a6fc3c70191769b8d07567f57",
+    "project_visibility": "public",
+    "ci_config_ref_uri": "gitlab.com/openpubkey/gl-test//.gitlab-ci.yml@refs/heads/main",
+    "ci_config_sha": "9898863c7dcd844a6fc3c70191769b8d07567f57",
+    "jti": "3a8b958c-0117-4d61-8d84-90b186efc4e7",
+    "iat": 1722189366,
+    "nbf": 1722189361,
+    "exp": 1722192966,
+    "iss": "https://gitlab.com",
+    "sub": "project_path:openpubkey/gl-test:ref_type:branch:ref:main",
+    "aud": "OPENPUBKEY-PKTOKEN:1234"
+},
+"signatures":[
+  {"protected":{
+    "alg": "GQ256",
+    "cic": "HVIF0m3zCwEsAZSFjTiyQFU982qF2UZXSpCE__F6IbE",
+    "jkt": "4i3sFE7sxqNPOT7FdvcGA1ZVGGI_r-tsDXnEuYT4ZqE",
+    "kid": "eyJraWQiOiI0aTNzRkU3c3hxTlBPVDdGZHZjR0ExWlZHR0lfci10c0RYbkV1WVQ0WnFFIiwidHlwIjoiSldUIiwiYWxnIjoiUlMyNTYifQ",
+    "typ": "JWT"
+    },
+  "signature":"18nN... (5504 base64 characters)"},
+  {"protected":{
+    "alg": "ES256",
+    "rz": "600e69b29d89651591836d2598f6813a9a74b9e4124ddb81bee1561299c3590e",
+    "typ": "CIC",
+    "upk": {
+        "alg": "ES256",
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "c63goURlnP5vbJbt4chtOHTHwg6Yvy4h6_aw3Zc2A5o",
+        "y": "pfsH8--s5c8u4DxXto0sN4g5n6SjlXn1WjzaKXrr9b4"
+    }
+  },
+  "signature":"xpt6tTEzAqPrXCzAL6UCdiIliPPejRmn2sW1_RSxCt3WUQN38s3z_MURF3Bd8mlncU1UhWwSMCAcwDijm-Hc6w"}
+]}
+```
+
+#### GQ-Commitment to the CIC
+
+For Google and github we bind the CIC (Client Instance Claims) that contains the user's public key to the ID Token by putting a hashed commitment to the CIC in one of the claims of the ID Token. For Google we put the commitment in the `nonce` claim, for github we put the commitment in the audience (`aud`) claim. Gitlab-CI does not provide an `nonce` claim and does not allow a running job/workflow to specify the audience. While Gitlab does allow customizing the audience, this custom audience is a fixed configuration value and can not be set per request. As such we can not use audience-commitment PK Tokens for gitlab-CI. As gitlab-CI is for machine identity it does not support specifying a nonce and thus nonce-commitment PK Tokens are not available either.
+
+GQ-binding solves this problem posed by gitlab-CI. Remember that with GQ signatures, we use the OP's RSA signature on the ID Token to generate a GQ signature that signs both the payload and the GQ protected header. Therefore, we can simply put the commitment to the identity's public key, i.e. the hash of the CIC, in the GQ protected header and then when the GQ signature will also sign this commitment. GQ-bound PK Tokens were introduced in PR: [Adds gitlab-ci OP using GQ commitment binding](https://github.com/openpubkey/openpubkey/pull/143).
```

Comment:
Not sure I'm following this last part of the sentence here "and then when the GQ signature will also sign this commitment." Do you mean we put the CIC hash in both places?

Reply:
Rewrote